src/entity/mixins/flowchart.js

```diff
@@ -0,0 +1,67 @@
+/* eslint-disable max-classes-per-file */
+import lodash from "lodash";
+
+import { addUnit, removeUnit, replaceUnit } from "../../utils";
+
+export const FlowchartEntityMixin = (superclass) => {
+    return class extends superclass {
+        constructor(config) {
+            super(config);
+            this._units = config.units || [];
+        }
+
+        get units() {
+            return this._units;
+        }
+
+        setUnits(units) {
+            this._units = units;
+        }
+
+        addUnit(unit, index = -1) {
+            this._units = addUnit(this.units, unit, index);
+        }
+
+        removeUnit(flowchartId) {
+            this._units = removeUnit(this.units, flowchartId);
+        }
+
+        replaceUnit(unit, index) {
+            this._units = replaceUnit(this.units, unit, index);
+        }
+
+        getUnit(flowchartId) {
+            return this.units.find((x) => x.flowchartId === flowchartId);
+        }
+
+        unitIndex(flowchartId) {
```

Comment:
Let's call this `getUnitIndexByFlowchartId` pls for clarity and update the subworkflow component accordingly:

```js
    onUnitUpdate(unit) {
        const subworkflow = this.state.entity;
        const index = subworkflow.unitIndex(unit.flowchartId);
        subworkflow.replaceUnit(index, unit);
        this._resetStateEntityAndUpdateParents(subworkflow);
    }
```